cppcache/src/TcrConnection.cpp

```diff
@@ -691,51 +691,70 @@ char* TcrConnection::sendRequest(const char* buffer, size_t len,
   return readMessage(recvLen, receiveTimeoutSec, true, &opErr, false, request);
 }
 
-void TcrConnection::sendRequestForChunkedResponse(
-    const TcrMessage& request, size_t len, TcrMessageReply& reply,
-    std::chrono::microseconds sendTimeoutSec,
-    std::chrono::microseconds receiveTimeoutSec) {
-  auto msgType = request.getMessageType();
-  switch (msgType) {
-    case TcrMessage::QUERY:
-    case TcrMessage::QUERY_WITH_PARAMETERS:
-    case TcrMessage::EXECUTECQ_WITH_IR_MSG_TYPE:
-    case TcrMessage::GETDURABLECQS_MSG_TYPE:
-    case TcrMessage::EXECUTE_FUNCTION:
-    case TcrMessage::EXECUTE_REGION_FUNCTION:
-    case TcrMessage::EXECUTE_REGION_FUNCTION_SINGLE_HOP: {
-      receiveTimeoutSec = reply.getTimeout();
-      sendTimeoutSec = reply.getTimeout();
-      break;
-    }
-    default:
-      break;
-  }
-
+std::chrono::microseconds TcrConnection::sendWithTimeouts(
+    const char* data, size_t len, std::chrono::microseconds sendTimeout,
+    std::chrono::microseconds receiveTimeout) {
   std::chrono::microseconds timeSpent{0};
-  send(timeSpent, request.getMsgData(), len, sendTimeoutSec, true);
+  send(timeSpent, data, len, sendTimeout, true);
 
-  if (timeSpent >= receiveTimeoutSec)
+  if (timeSpent >= receiveTimeout) {
     throwException(
         TimeoutException("TcrConnection::send: connection timed out"));
+  }
 
-  receiveTimeoutSec -= timeSpent;
+  return timeSpent;
+}
+
+bool TcrConnection::replyHasResult(const TcrMessage& request,
+                                   TcrMessageReply& reply) {
+  auto hasResult = true;
 
-  // to help in decoding the reply based on what was the request type
-  reply.setMessageTypeRequest(msgType);
   // no need of it now, this will not come here
-  if (msgType == TcrMessage::EXECUTE_REGION_FUNCTION_SINGLE_HOP) {
+  if (request.getMessageType() ==
+      TcrMessage::EXECUTE_REGION_FUNCTION_SINGLE_HOP) {
     ChunkedFunctionExecutionResponse* resultCollector =
         static_cast<ChunkedFunctionExecutionResponse*>(
             reply.getChunkedResultHandler());
     if (resultCollector->getResult() == false) {
       LOGDEBUG(
           "TcrConnection::sendRequestForChunkedResponse: function execution, "
           "no response desired");
-      return;
+      hasResult = false;
     }
   }
-  readMessageChunked(reply, receiveTimeoutSec, true);
+
+  return hasResult;
+}
+
+void TcrConnection::sendRequestForChunkedResponse(
```

Comment:
complete and total nitpick... maybe next time put the all the extracted methods below the parent method, just to end with a more digestible Diff... i.e. I think the changes would at least difference from block of code to new method call etc... e.g. the `switch(msgType)` block would diff to the call of `replayHasValidTimeout()`

Reply:
ima try moving that and see how it affects the diff.  Thanks for this feedback, it's always a tough call to know exactly where to put new methods.